From ea0add0a7f0da3df62b8827189185d34f0416585 Mon Sep 17 00:00:00 2001
From: Sam Nelson <sam.nelson@ti.com>
Date: Mon, 14 Sep 2015 15:56:43 -0400
Subject: [PATCH 05/12] Documentation: uio-ti-pdrv: Add documentation for
 ti,uio-ti-pdrv

- uio-ti-pdrv add user space access to TI peripheral modules

Signed-off-by: Sam Nelson <sam.nelson@ti.com>
---
 .../bindings/arm/keystone/ti,uio-ti-pdrv.txt       | 63 ++++++++++++++++++++++
 1 file changed, 63 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/arm/keystone/ti,uio-ti-pdrv.txt

diff --git a/Documentation/devicetree/bindings/arm/keystone/ti,uio-ti-pdrv.txt b/Documentation/devicetree/bindings/arm/keystone/ti,uio-ti-pdrv.txt
new file mode 100644
index 0000000..1fb68a0
--- /dev/null
+++ b/Documentation/devicetree/bindings/arm/keystone/ti,uio-ti-pdrv.txt
@@ -0,0 +1,63 @@
+UIO ti platform driver
+
+This file provides information about, what the device tree entry for the
+TI uio accesible peripheral module devices should contain.
+
+Required properties:
+- compatible : "ti,uio-ti-pdrv"
+- reg : Register start address and the size that will be
+		used by the driver
+ or
+  mem : Register/Memory start address and size of the Register/memory region.
+
+Recommended properties :
+- interrupts : 		Standard interrupt property for interrupt
+			generated from the peripheral module  to the host
+			processor.
+			First entry:
+
+- clocks : 		desired clock domain property
+			For example can be defined as per the bindings in
+			Documentation/devicetree/bindings/clock/keystone-gate.txt
+
+Optional properties:
+- cfg-params: Specifies configuration parameters to be used with specific peripherals
+	ti,serdes_refclk_khz: serdes reference clock in khz
+	ti,serdes_maxserrate_khz: serdes maximum serial rate in khz
+	ti,serdes_maxlanerate: Lane rate "half" or "full"
+	ti,serdes_numlanes: Number of lanes
+	The following are serdes equilization tap parameters
+	ti,serdes_c1	: Serdes c1 coefficient (0-31), one per lane
+	ti,serdes_c2	: Serdes c2 coefficient (0-15), one per lane
+	ti,serdes_cm	: Serdes cm coefficient (0-15), one per lane
+	ti,serdes_tx_att: Transmit att, one per lane
+	ti,serdes_tx_vreg: Transmit vreg, one per lane
+	ti,serdes_rx_att: Receive att, one per lane
+	ti,serdes_rx_boost: Receive boost, one per lane
+	ti,qm-queue     : QMSS queue associated with qpend binding
+
+Example:
+
+	uio_hyperlink0: hyperlink0 {
+		compatible = "ti,uio-ti-pdrv";
+		mem  = <0x21400000 0x00000100
+			0x40000000 0x10000000
+			0x0231a000 0x00002000>;
+		clocks = <&clkhyperlink0>;
+		interrupts	= <0 387 0x101>;
+		label = "hyperlink0";
+		cfg-params
+		{
+			ti,serdes_refclk_khz = <312500>;
+			ti,serdes_maxserrate_khz = <6250000>;
+			ti,serdes_lanerate = "half";
+			ti,serdes_numlanes = <4>;
+			ti,serdes_c1 = <4 4 4 4>;
+			ti,serdes_c2 = <0 0 0 0>;
+			ti,serdes_cm = <0 0 0 0>;
+			ti,serdes_tx_att = <12 12 12 12>;
+			ti,serdes_tx_vreg = <4 4 4 4>;
+			ti,serdes_rx_att = <11 11 11 11>;
+			ti,serdes_rx_boost = <3 3 3 3>;
+		};
+	};
-- 
1.9.1

